deepcell_toolbox/multiplex_utils_test.py

```diff
@@ -0,0 +1,169 @@
+    processed_thresh = multiplex_preprocess(img, normalize=False)
+    assert not (processed_thresh <= 1).all()
+
+    new_spot_val = processed_thresh[0, 200, 200, 0]
+    processed_thresh[0, 200, 200, 0] = 0.5
+    next_max_val = np.max(processed_thresh)
+    assert np.round(new_spot_val / next_max_val, 1) == 1
+
+    # no change to image
+    not_processed = multiplex_preprocess(img, normalize=False, threshold=False)
+    assert np.all(not_processed == img)
+
+
+def test_multiplex_postprocess(mocker):
+    # create dict, with each image having a different constant value
+    base_array = np.ones((1, 20, 20, 1))
+
+    whole_cell_list = [base_array * mult for mult in range(1, 3)]
+    whole_cell_dict = {'inner-distance': whole_cell_list[0],
+                       'pixelwise-interior': whole_cell_list[1]}
+
+    nuclear_list = [base_array * mult for mult in range(3, 5)]
+    nuclear_dict = {'inner-distance': nuclear_list[0],
+                    'pixelwise-interior': nuclear_list[1]}
+
+    model_output = {'whole-cell': whole_cell_dict, 'nuclear': nuclear_dict}
+
+    # whole cell predictions only
+    whole_cell = multiplex_postprocess(model_output=model_output,
+                                       compartment='whole-cell')
+    assert whole_cell.shape == (1, 20, 20, 1)
+
+    # nuclear predictions only
+    nuclear = multiplex_postprocess(model_output=model_output,
+                                    compartment='nuclear')
+    assert nuclear.shape == (1, 20, 20, 1)
+
+    # both whole-cell and nuclear predictions
+    both = multiplex_postprocess(model_output=model_output,
+                                 compartment='both')
+    assert both.shape == (1, 20, 20, 2)
+
+    # make sure correct arrays are being passed to helper function
+    def mock_deep_watershed_mibi(model_output):
+        pixelwise_interior_vals = model_output['pixelwise-interior']
+        return pixelwise_interior_vals
+
+    mocker.patch('deepcell_toolbox.deep_watershed.deep_watershed_mibi',
+                 mock_deep_watershed_mibi)
+
+    # whole cell predictions only
+    whole_cell_mocked = multiplex_postprocess(model_output=model_output,
+                                              compartment='whole-cell')
+
+    assert np.array_equal(whole_cell_mocked, whole_cell_dict['pixelwise-interior'])
+
+    # nuclear predictions only
+    whole_cell_mocked = multiplex_postprocess(model_output=model_output,
+                                              compartment='nuclear')
+
+    assert np.array_equal(whole_cell_mocked, nuclear_dict['pixelwise-interior'])
+
+    with pytest.raises(ValueError):
+        whole_cell = multiplex_postprocess(model_output=model_output,
+                                           compartment='invalid')
+
+
+def test_format_output_multiplex():
+
+    # create output list, each with a different constant value across image
+    base_array = np.ones((1, 20, 20, 1))
+
+    whole_cell_list = [base_array * mult for mult in range(1, 8)]
```

Comment:
And the test function too
https://github.com/vanvalenlab/deepcell-tf/blob/f1838ea700ea063e8f28c27da019e13526c43258/deepcell/applications/multiplex_segmentation_test.py#L87